### ISSUE TITLE
Trying to set up network for regression task

### ISSUE THREAD

**Issue by emanuelazcona** [tkipf/kegra#34]
Could you possibly point me on where to start with this so I can set this up for a regression task? Plan on using L2_loss.

I have multiple samples of data that lie on the same graph, however the features at each node is a vector. I was wondering how to get around since I'm dealing with a 3D tensor:

X : (Samples x Nodes x features)
Y: (Samples x Nodes x features) as well

Most of the operations in both implementations seem to only work for X being a matrix, unless I'm mistaken.

Of course, my adjacency & laplacian will remain 2D matrices for all operations (there the Cheby. will be too)

**Comment by tkipf** [comment]
For regression, simply swap out the loss function (L2 instead of cross
entropy should do fine).

If you have multiple separate graph instances, just concatenate their
feature matrices. You can accordingly ‘concatenate’ the adjacency matrices
into a large N_total x N_total matrix with block-diagonal structure.
N_total is the total number of nodes after concatenation.

**Comment by emanuelazcona** [comment]
Thanks for the tip! Will try now.

Also, have you guys worked on pooling for this? I know Defferrard et al. built their Chebynet with pooling. Just wondering if there is a Keras implementation yet.

**Comment by tkipf** [comment]
Have a look here at my previous works:
https://scholar.google.com/citations?user=83HL5FwAAAAJ

Aside from global gated pooling (in the recent MolGAN paper) and some form
of self-attentive pooling (in the NRI paper), we haven’t done much along
these lines.
